services/rds/src/main/java/software/amazon/awssdk/services/rds/RdsUtilities.java

```diff
@@ -0,0 +1,139 @@
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.rds;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.function.Consumer;
+import software.amazon.awssdk.annotations.Immutable;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.signer.Aws4Signer;
+import software.amazon.awssdk.auth.signer.params.Aws4PresignerParams;
+import software.amazon.awssdk.awscore.client.config.AwsClientOption;
+import software.amazon.awssdk.core.client.config.SdkClientConfiguration;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+import software.amazon.awssdk.http.SdkHttpMethod;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.rds.model.GenerateAuthenticationTokenRequest;
+import software.amazon.awssdk.utils.StringUtils;
+
+@Immutable
+@SdkPublicApi
+public final class RdsUtilities {
+    private final SdkClientConfiguration clientConfiguration;
+
+    /**
+     * SDK currently validates that region is present while constructing {@link RdsUtilities} object.
+     * This can be relaxed in the future when more methods are added that don't use region.
+     */
+    private RdsUtilities(Builder builder) {
+        this.clientConfiguration = builder.clientConfiguration;
+    }
+
+    /**
+     * Creates a builder for {@link RdsUtilities}.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    // Used by low-level client
+    @SdkInternalApi
+    static RdsUtilities create(SdkClientConfiguration clientConfiguration) {
+        return RdsUtilities.builder().clientConfiguration(clientConfiguration).build();
+    }
+
+    public String generateAuthenticationToken(Consumer<GenerateAuthenticationTokenRequest.Builder> getUrlRequest) {
+        return generateAuthenticationToken(GenerateAuthenticationTokenRequest.builder().applyMutation(getUrlRequest).build());
+    }
+
+    public String generateAuthenticationToken(GenerateAuthenticationTokenRequest request) {
+        SdkHttpFullRequest httpRequest = SdkHttpFullRequest.builder()
+                                            .method(SdkHttpMethod.GET)
+                                            .protocol("https")
+                                            .host(request.hostname())
+                                            .port(request.port())
+                                            .encodedPath("/")
+                                            .putRawQueryParameter("DBUser", request.userName())
+                                            .putRawQueryParameter("Action", "connect")
+                                            .build();
+
+        Instant expirationTime = Instant.now().plus(15, ChronoUnit.MINUTES);
```

Comment:
This probably needs a link to the doc where it states this must be 15 min

Reply:
Ya, our doc pages all say 15 mins: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.html